pkg/storage/s3/s3.go

```diff
@@ -211,35 +260,69 @@ func (d *driver) CreateStorage(cr *opapi.ImageRegistry, modified *bool) error {
 		return err
 	}
 
-	for i := 0; i < 5000; i++ {
-		if len(d.Config.Bucket) == 0 {
-			d.Config.Bucket = fmt.Sprintf("%s-%s-%s-%s", clusterconfig.StoragePrefix, d.Config.Region, strings.Replace(string(cv.Spec.ClusterID), "-", "", -1), strings.Replace(string(uuid.NewUUID()), "-", "", -1))[0:62]
-		}
-
-		_, err := svc.CreateBucket(&s3.CreateBucketInput{
-			Bucket: aws.String(d.Config.Bucket),
-		})
+	// If a bucket name is supplied, and it already exists and we can access it
+	// just update the config
+	var bucketExists bool
+	if len(d.Config.Bucket) != 0 {
+		bucketExists, err = d.bucketExists(d.Config.Bucket)
 		if err != nil {
 			if aerr, ok := err.(awserr.Error); ok {
 				switch aerr.Code() {
-				case s3.ErrCodeBucketAlreadyExists:
-					if cr.Spec.Storage.S3.Bucket != "" {
-						break
-					}
-					d.Config.Bucket = ""
-					continue
-				default:
+				case s3.ErrCodeNoSuchBucket, "Forbidden", "NotFound":
+					// If the bucket doesn't exist that's ok, we'll try to create it
 					util.UpdateCondition(cr, opapi.StorageExists, operatorapi.ConditionFalse, aerr.Code(), aerr.Error(), modified)
+				default:
+					util.UpdateCondition(cr, opapi.StorageExists, operatorapi.ConditionUnknown, "Unknown Error Occurred", err.Error(), modified)
 					return err
 				}
 			}
 		}
-		break
 	}
+	if len(d.Config.Bucket) != 0 && bucketExists {
+		*cr.Status.Storage.S3 = *d.Config
```

Comment:
assuming you're trying to avoid referencing the same thing, you need to do a deepcopy here.  Otherwise any pointer fields embedded in these objects will still point to the same place.

Reply:
updated